blueprints/data-solutions/dataproc-lifecycle-via-composer/README.md

```diff
@@ -0,0 +1,115 @@
+
+This approach aims to use resources efficiently meanwhile minimize provisioning and execution time.
+
+This is the high level diagram:
+![](images/dataproc_lifecycle.png)
+
+
+## Prerequisites
+1. This blueprint will deploy all its resources into the project defined by the `project_id` variable. Please note that we assume this project already exists.
+2. The user deploying the project _(executing terraform plan/apply)_ should have admin permissions in the selected project, or permission to create all the resources in the Terraform **main.tf** script. 
+
+## Project Folder Structure
+ ```bash
+ terraform/
+├── main.tf
+├── variables.tf 
+dags/          (Autogenerated on Terraform Plan/Apply from /dag_config/ files)
+├── ephemeral_cluster_job_1.py
+├── ephemeral_cluster_job_2.py
+jobs/
+├── hello_world_spark.py
+├── ...        (Add your dataproc jobs here)
+include/
+├── dag_template.py
+├── generate_dag_files.py
+└── dag_config
+    ├── dag1_config.json
+    └── dag2_config.json
+    └── ...     (Add your Composer/Airflow DAGs configuration here)
+  
+ ```
+## Adding Jobs
+#### Prepare Dataproc Jobs to be executed
+1. Clone this repository
+2. Locate your Dataproc jobs in the **/jobs/** folder in your local environment
+
+#### Prepare Composer Dags to be deployed
+3. Locate your DAG configuration files in the **/include/dag_config/** folder in your local environment. DAG configuration files have the following variables:
+```json
+{
+    "DagId": "ephemeral_cluster_job_1",     --DAG name you will see in Airflow environment
+    "Schedule": "'@daily'",                 --DAG Schedule
+    "ClusterName":"ephemeral-cluster-test", --Dataproc Cluster to be Used/created for this DAG/Job to be executed in
+    "StartYear":"2022",                     --DAG start year
+    "StartMonth":"9",                       --DAG start month
+    "StartDay":"13",                        --DAG start day
+    "Catchup":"False",                      --DAG backfill catchup
+    "ClusterMachineType":"n1-standard-4",   --Dataproc machine type to be used by master and worker cluster nodes
+    "ClusterIdleDeleteTtl":"300",           --Time in seconds to delete unused Dataproc cluster
+    "SparkJob":"hello_world_spark.py"       --Spark Job to be executed by DAG, should be placed in /jobs/ folder of this project. (if other type of Dataproc jobs modify dag_template.py)
+}
+```
+4. (Optional) You can run `python3 ../include/generate_dag_files.py` in your local environment, if you want to review generated DAGs before deploying(TF plan/apply) those.
+
+## Deployment
+
+1. Set Google Cloud Platform credentials on local environment: 
+https://cloud.google.com/source-repositories/docs/authentication
+
+
+2. You must supply the following terraform variables in order to deploy the project:
```

Comment:
Variables you can configured are already mentioned in the dedicated section. I would list here only the minimum set of variables to be configured.

Reply:
done